messagecatalog/spec.md

```diff
@@ -0,0 +1,1042 @@
+the authority is a person or group of people or a system that controls the
+message definition. For the most common cases, the authority is whoever manages
+the message definition in the catalog.
+
+For the replication model, the authority information MUST be used to
+disambiguate message definitions from different origins that need to be
+consolidated in central catalogs or in caches.
+
+In complex business systems, the authority might be centralized and message
+definitions are explicitly designed, reviewed and approved for use. In simpler
+scenarios, the authority might simply lay with any producer of events, and
+message definitions might be inferred from code artifacts without developers
+being aware.
+
+The [authority](#authority-definition-authority) attribute of the message
+definition object reflects the controlling entity. The
+[authority](#authority-definition-authority) attribute MAY be set to any URI.
+The URI does not have to correspond to a resolvable network endpoint, even if a
+URI scheme like "http" is used in order to borrow its generally well-understood
+structure. If the URI does not correspond to an active network endpoint,
+ownership rights of corresponding DNS domain name owners SHOULD nevertheless be
+respected.
+
+For centralized governance scenarios, the `authority` SHOULD be set to a URI
+reflecting the governing entity, like `https://messagecatalog.corp.example.com`.
+
+### 2.2.2. Message definition version URI
+
+To refer to a specific message definition object, unambiguous references to
+specific [definition versions](#22-schema-and-schema-version) are needed. These
+references can be resolved into definition documents with the help of a catalog.
+
+The message definition version URI is composed from the
+[authority](#authority-definition-authority) (as the base URI) and the
+[message definition version id](#id-definition-version-id).
+
+In the default case, this URI corresponds directly to the URI of the
+["getMessage definitionVersion"](#342-get-a-specific-schema-version) API
+operation, which is convenient for simple usage scenarios.
+
+In [replication scenarios](#4-replication-model-and-state-change-events), the
+URI might not correspond to a resolvable network address, as permitted for the
+authority URI in the previous section. A consumer SHOULD then be configured with
+a fixed message catalog endpoint for it to use, and obtain definitions
+identified by a URI using the
+["getMessage definitionVersionByURI"](#342-get-a-specific-schema-version) API
+operation on that endpoint, rather than trying to resolve the URI directly.
+
+The following table shows some examples for a catalog hosted at
+`https://example.com/messagecatalog`:
+
+| [authority](#authority-definition-authority) | [id](#id-definition-version-id)                               | Message definition URI                                                                         |
```

Comment:
Is `Message definition URI` and `Message definition version URI` the same?

Reply:
What is a `definition version id`? Below an `id` is defined for a _message definition_, and a `version` is defined for a _message definition version_.